lib/versions.py

```diff
@@ -86,23 +40,67 @@ def is_centos():
     def __bool__():
         return _os_release['ID'] in ['rhel', 'centos']
 
+    @staticmethod
+    def _parse_version(version):
+        if isinstance(version, _Rhel):
+            return version
+        if not isinstance(version, str):
+            version = str(version)
```

Comment:
This is probably fine, but consider using integers everywhere and just do `str -> int` conversion when reading from `os-release` (or interfacing with `rpm` via `subprocess`), just to avoid the extra str/int confusion in classes.